rancher2/structure_app.go

```diff
@@ -158,5 +161,9 @@ func expandApp(in *schema.ResourceData) (*projectClient.App, error) {
 		obj.Labels = toMapString(v)
 	}
 
+	if v, ok := in.Get("timeout").(int); ok && v > 0 {
+		obj.Timeout = int64(v)
+	}
```

Comment:
To use terraform operation timeout, obj.Timeout should get the value from resource

```
-	if v, ok := in.Get("timeout").(int); ok && v > 0 {
-		obj.Timeout = int64(v)
-	}
+ obj.Timeout = int64(in.Timeout(schema.TimeoutCreate).Seconds())
```